### DISCUSSION TITLE
Locust Login Request Failed

### DISCUSSION THREAD

**Discussion by nicholaswkc34** [locustio/locust#2802] [General]
Dear all forumers, 

I started a project using Locust for login request but it failed which shows on the report. When i click the login button, the url is as show in the code. 
I have the host as https://staging-ticket.revid.my/. Then my code is as follow: 

```
from locust import HttpUser, task, between

class WebsiteUser(HttpUser):
    wait_time = between(1, 2)

    @task
    def homepage(self):
        self.client.get("/")

    @task
    def login(self):
        self.client.post("https://stg-accounts.mediaprimaplus.com.my/authorize?client_id=632354224783531&redirect_uri=https%3A%2F%2Fstaging-ticket.revid.my%2Fcallback&response_type=code&state=%2F",
                         {"username":"user", "password":"password" })
```

What could be wrong with my code? 
Please help. Im totaly new to Locust. Please bear with me.

**Comment by cyberw** [comment]
I dont see anything obviously wrong with your code. Maybe it expects json instead of a urlencoded/form-data? Then you wanna use `json={"username":"user", "password":"password" }`

To make `python-requests` print the exact request sent, see https://docs.locust.io/en/stable/running-in-debugger.html#print-http-communication

That makes it easier to compare locust's request to what the browser does (I'm assuming it works in browser).

**Comment by nicholaswkc34** [comment]
Let me add the logging facility and see how it's going.

**Comment by nicholaswkc34** [comment]
The login post request does not have any error now but there is new get request which require login then it failed. This is the error i get from report. 

29 | GET | /account?tab=order | HTTPError('500 Server Error: Internal Server Error for url: /account?tab=order')
32 | GET | /account?tab=account | HTTPError('500 Server Error: Internal Server Error for url: /account?tab=account')

These two requests require to login first before it appear on web browsers UI. Please help. How to test it? 

Besides, I have error in too many requests sent in short time frame. How to solve this?
5880 | GET | / | HTTPError('429 Client Error: Too Many Requests for url: /')

**Comment by nicholaswkc34** [comment]
Please help me.Anyone can answers the questions? Thanks a lot.

**Comment by cyberw** [comment]
Did you log the request as it was sent by Locust using the suggestion in my first reply and compare it to what your browser sent?

The 429 is your server (or load balancer) blocking you, you need to investigate on your server side. Its unlikely you can solve this by modifying your requests/scenario.

**Comment by nicholaswkc34** [comment]
I can added the code from documentation, How to compare the between Locust and browser sent? Sorry to asking stupid question.

**Comment by cyberw** [comment]
You can use chrome developer tools to show the traffic sent by the browser. Here's my first hit on google :) https://docs.coveo.com/en/2146/coveo-for-sitecore-v5/analyze-http-traffic-using-chrome-developer-tools
